next_sparseconvnet/utils/train_utils.py

```diff
@@ -0,0 +1,82 @@
+import numpy as np
+import torch
+import sys
+import sparseconvnet as scn
+from .data_loaders import DataGen, collatefn, LabelType
+from next_sparseconvnet.networks.architectures import UNet
+
+def IoU(true, pred, nclass = 3):
+    """
+        Intersection over union is a metric for semantic segmentation.
+        It returns a IoU value for each class of our input tensors/arrays.
+    """
+    eps = sys.float_info.epsilon
+    confusion_matrix = np.zeros((nclass, nclass))
+
+    for i in range(len(true)):
+        confusion_matrix[true[i]][pred[i]] += 1
+
+    IoU = []
+    for i in range(nclass):
+        IoU.append((confusion_matrix[i, i] + eps) / (sum(confusion_matrix[:, i]) + sum(confusion_matrix[i, :]) - confusion_matrix[i, i] + eps))
+    return IoU
+
+
+def train_one_epoch(epoch_id, net, criterion, optimizer, loader):
+    """
+        Trains the net for all the train data one time
+    """
+    net.train()
+    loss_epoch, iou_epoch = [], []
+    for batchid, (coord, ener, label, event) in enumerate(loader):
+        label = label.type(torch.LongTensor) #quitar esto una vez se corrija en el collate
+        batch_size = len(event)
+        ener, label = ener.cuda(), label.cuda()
+
+        optimizer.zero_grad()
+
+        output = net.forward((coord, ener, batch_size))
+
+        loss = criterion(output, label)
+        loss.backward()
+
+        optimizer.step()
+
+        loss_epoch.append(loss.item())
+
+        #IoU
+        softmax = torch.nn.Softmax(dim = 1)
+        prediction = torch.argmax(softmax(output), 1)
+        iou_epoch.append(IoU(label.cpu(), prediction.cpu()))
+
+        if batchid%2==0:
```

Comment:
The 2 for sure should not be hardcoded. In most cases we just want to know the values at the end of the epoch, but if you prefer you can do something like adding new parameter `print_every = None`, and if is defined then print result every `print_every steps`, other simply print at the end of the epoch?